tests/unit/test_ca.cpp

```diff
@@ -222,70 +239,162 @@ TEST_F(CATest, testIterateOnExtensionMap)
     EXPECT_EQ(2, _signParams.extensionMap().size());
 }
 
-TEST_F(CATest, testCreateRootCertificate)
+TEST_F(CATest, testCreateRootRSACertificate)
 {
-    testValiditySpan(*_rootCert, _signParams.certificateValidity(),
+    testValiditySpan(*_rootRsaCert, _signParams.certificateValidity(),
                      _rootCertSignPoint);
-    EXPECT_EQ(*_rootCertDetails, _rootCert->getIssuerDistinguishedName());
-    EXPECT_EQ(*_rootCertDetails, _rootCert->getSubjectDistinguishedName());
-    EXPECT_EQ(*_rootKey, _rootCert->getPublicKey());
-    EXPECT_NO_THROW(_rootCert->verify({*_rootCert}, {}));
+    EXPECT_EQ(*_rootCertDetails, _rootRsaCert->getIssuerDistinguishedName());
+    EXPECT_EQ(*_rootCertDetails, _rootRsaCert->getSubjectDistinguishedName());
+    EXPECT_EQ(*_rootRSAKey, _rootRsaCert->getPublicKey());
+    EXPECT_NO_THROW(_rootRsaCert->verify({*_rootRsaCert}, {}));
 }
 
-TEST_F(CATest, testCAContents)
+TEST_F(CATest, testCreateRootECCCertificate)
 {
-    EXPECT_EQ(_rootCert->toPEM(), _ca->getRootCertificate().toPEM());
-    EXPECT_EQ(_signParams, _ca->getSignParams());
+    testValiditySpan(*_rootEccCert, _signParams.certificateValidity(),
+                     _rootCertSignPoint);
+    EXPECT_EQ(*_rootCertDetails, _rootEccCert->getIssuerDistinguishedName());
+    EXPECT_EQ(*_rootCertDetails, _rootEccCert->getSubjectDistinguishedName());
+    EXPECT_EQ(*_rootEccKey, _rootEccCert->getPublicKey());
+    EXPECT_NO_THROW(_rootEccCert->verify({*_rootEccCert}, {}));
 }
 
-TEST_F(CATest, testSignedCSRHasCorrectFields)
+TEST_F(CATest, testRSACAContents)
+{
+    EXPECT_EQ(_rootRsaCert->toPEM(), _rsaCa->getRootCertificate().toPEM());
+    EXPECT_EQ(_signParams, _rsaCa->getSignParams());
+}
+
+TEST_F(CATest, testECCCAContents)
+{
+    EXPECT_EQ(_rootEccCert->toPEM(), _eccCa->getRootCertificate().toPEM());
+    EXPECT_EQ(_signParams, _eccCa->getSignParams());
+}
+
+TEST_F(CATest, testSignedCSRHasCorrectFieldsRSA)
```

Comment:
Maybe we could do this with an INSTANCIATE_TESTCASE as well?
Also for a lot of the above test cases.

Reply:
Agreed. Reworked the test_ca to use parametrized tests.